### DISCUSSION TITLE
Minio + Upload Metadata + Object?

### DISCUSSION THREAD

**Discussion by alexpebody** [minio/minio#12212] [Q&A]
Hi everyone! Guys, anybody know, maybe Minio not result and best product for it, but i need upload many objects and metadata and then i need can find this objects over metadata and get uri? I had install Monio server on Ubuntu 18.04 and i can't unserstand how i can do this func?

I need many files upload with metadata like position num, size, weight and other metadata of object; I need good performance like 1000 objects per hour; Upload with REST in URI data or the same.

Maybe some system already have this functions, like Hadoop or Minio or other?

Anybody know solved? Thank you so much.

As i know, Minio have MINIO SDK where i seen: opts minio.PutObjectOptions Allows user to set optional custom metadata, content headers, encryption keys and number of threads for multipart upload operation. But how example upload object to Minio with metadata? 

Maybe have some better product with opensorce or box product for this task? Any way, any example? Maybe this will can solved with python or some SDK API Minio?

Thanks for all! 8-)

**Comment by alexpebody** [comment]
Any example, how upload object with metadata? Please.

**Comment by klauspost** [comment]
See https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html on how to add custom metadata.

The performance requirements generally shouldn't be a problem on decent hardware in a lightly distributed setup.  You can use warp to benchmark your setup: https://github.com/minio/warp

If you need more detailed help, we offer that as part of our subscription package: https://min.io/pricing

**Comment by alexpebody** [comment]
So thx. I found this:
https://github.com/minio/minio-js/blob/master/examples/fput-object.js
https://github.com/minio/minio-py/blob/3d01781eb0fc14857e1e11014483cdfd0186cb36/tests/functional/tests.py#L316-L334
https://github.com/minio/minio/blob/9aeedc4c5f327e6299382d6238f8fcee03e23b9b/cmd/object-api-datatypes.go#L149-L153

How i can simple using this, i am not a programmist... 8-(

**Comment by alexpebody** [comment]
Is there any example of implementation to put an object with metadata and get an object using metadata? THANK YOU!

**Reply by kannappanr** [reply]
@alexpebody https://github.com/minio/minio-js/blob/master/examples/fput-object.js uploads an object and metadata. There is no way to download an object based on it's metadata in AWS S3 compatible object stores.
